functional-tests/src/test/groovy/specs/CreditTransactionSpec.groovy

```diff
@@ -51,7 +51,9 @@ class CreditTransactionSpec extends LoggedInSpec {
       at new ToastModal('Success!', 'Credit transaction recommended.')
     and: 'My unread notification count has increased by 1'
       page(HomePage)
-      headerModule.getNotificationCount() == initialNotificationCount+1
+      waitFor {
+        headerModule.getNotificationCount() == initialNotificationCount+1
```

Comment:
HeaderModule.groovy # getNotificationCount() already has a wait built in, which defaults to 3 seconds.

In the case where the notification count does not exist (because they have no unread notifications), then a traditional waitFor{} works great, since we can wait for the element to appear and then grab the value.

But in the case where the notification count already exists, but is being updated, we can't rely on waitFor{} since it would see the element and grab it immediately, getting the old value.  So we have to resort to a fixed sleep of some number of seconds.  

So our only consistent solution is to use a fixed sleep, as in the getNotificationCount() method.

But you can pass any integer to getNotificationCount() to increase or decrease it.